src/feature_extraction/base_features/ellipse_fitting.py

```diff
@@ -0,0 +1,65 @@
+import cv2
+import numpy as np
+
+from src.pose_estimation import PoseEstimation
+from src.feature_extraction.feature_base_class import Feature
+
+
+class EllipseFit(Feature):
+    """Feature for the best fit ellipse from the segmentation contours.
+    """
+
+    _name = 'ellipse fit'
+    _feature_names = ['x', 'y', 'a', 'b', 'c', 'w', 'l', 'theta']
+
+    def __init__(self, poses: PoseEstimation, pixel_scale: float):
+        super().__init__(poses, pixel_scale)
+
+    def per_frame(self, identity: int) -> np.ndarray:
+        values = np.zeros((self._poses.num_frames, len(self._feature_names)), dtype=np.float32)
+
+        seg_data = self._poses.get_segmentation_data(identity)
+
+        x = self.feature_names.index('x')
+        y = self.feature_names.index('y')
+        a = self.feature_names.index('a')
+        b = self.feature_names.index('b')
+        c = self.feature_names.index('c')
+        w = self.feature_names.index('w')
+        ln = self.feature_names.index('l')
+        t = self.feature_names.index('theta')
+
+        for frame in range(values.shape[0]):
+            contours = seg_data[frame, ...]
+            contours = contours.reshape(
+                contours.shape[0] * contours.shape[1], contours.shape[-1]
+                ).astype(np.int32)
+
+            moments = cv2.moments(

```

Comment:
TC-307, need to make sure pixel_scaling is factored in here so that the units (length) are correct after mathematical calculations.